app/src/main/java/org/thoughtcrime/securesms/database/LokiMessageDatabase.kt

```diff
@@ -213,32 +213,34 @@ class LokiMessageDatabase(context: Context, helper: SQLCipherOpenHelper) : Datab
         }
     }
 
-    fun getMessageServerHash(messageID: Long): String? {
-        val database = databaseHelper.readableDatabase
-        return database.get(messageHashTable, "${Companion.messageID} = ?", arrayOf(messageID.toString())) { cursor ->
+    fun getMessageServerHash(vararg keys: Long): String? = keys.asSequence().firstNotNullOfOrNull {
+        databaseHelper.readableDatabase.get(messageHashTable, "${Companion.messageID} = ?", arrayOf(it.toString())) { cursor ->
             cursor.getString(serverHash)
         }
     }
 
-    fun setMessageServerHash(messageID: Long, serverHash: String) {
-        val database = databaseHelper.writableDatabase
-        val contentValues = ContentValues(2)
-        contentValues.put(Companion.messageID, messageID)
-        contentValues.put(Companion.serverHash, serverHash)
-        database.insertOrUpdate(messageHashTable, contentValues, "${Companion.messageID} = ?", arrayOf(messageID.toString()))
+    fun setMessageServerHash(timestamp: Long, serverHash: String) {
+        val content = ContentValues(2).apply {
+            put(Companion.messageID, messageID)
```

Comment:
This should probably be `put(Companion.messageID, timestamp)` right?

Reply:
absolutely